circe/src/main/scala/org/http4s/circe/middleware/JsonDebugErrorHandler.scala

```diff
@@ -34,27 +33,26 @@ object JsonDebugErrorHandler {
           messageFailureLogger.debug(mf)(
             s"""Message failure handling request: ${req.method} ${req.pathInfo} from ${req.remoteAddr
               .getOrElse("<unknown>")}""")
-          mf.toHttpResponse[F](req.httpVersion).map { firstResp =>
-            Response[F](
+          mf.inHttpResponse[F, G](req.httpVersion).map { firstResp =>
+            Response[G](
               status = firstResp.status,
               httpVersion = firstResp.httpVersion,
               headers = firstResp.headers.redactSensitive(redactWhen)
-            ).withEntity(JsonErrorHandlerResponse[F](req, mf))
+            ).withEntity(JsonErrorHandlerResponse[G](req, mf))
           }
-        case scala.util.control.NonFatal(t) =>
+        case t =>
```

Comment:
Why did we lose the `NonFatal` here?

Reply:
Fatal errors should never be bubbled up into the cats handleErrorWith in the first place.

Reply:
Okay, that's true. Not really awake yet.  This is just rendering what's already been raised, not catching things.  And unless someone already messed up, it won't be raised.